OJS.Workers.ExecutionStrategies/RunSpaAndExecuteMochaTestsExecutionStrategy.cs

```diff
@@ -242,8 +301,44 @@ protected override IExecutor CreateExecutor()
             IExecutionContext<TestsInputModel> executionContext,
             TestContext test)
         {
-            var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, test.Input);
-            return this.ExtractTestResultsFromReceivedOutput(processExecutionResult.ReceivedOutput, test.Id);
+            var preExecutionResult = this.Execute(executionContext, executor, this.PreExecuteCodeSavePath, test.Input);
+            Match match = Regex.Match(preExecutionResult.ReceivedOutput, @"Container port: (\d+);Container name: ([a-zA-Z-_]+);");
+            if (match.Success)
+            {
+                this.PortNumber = int.Parse(match.Groups[1].Value);
+                this.ContainerName = match.Groups[2].Value;
+
+                var filePath = FileHelpers.BuildPath(this.TestsPath, $"{test.Id}{JavaScriptFileExtension}");
+
+                // pass in container name in order to close container after execution
+                // pass test file path to mocha so it executes only this test file, and not all test files each run
+                var processedPythonCodeTemplate = this.PythonCodeTemplate
+                    .Replace(ContainerNamePlaceholder, this.ContainerName)
+                    .Replace(TestFilePathPlaceholder, filePath);
+
+                codeSavePath = this.SavePythonCodeTemplateToTempFile(processedPythonCodeTemplate);
```

Comment:
make codeSavePath local variable

Reply:
Extracted codeSavePath to a better named local variable and changed properties used only in this method to local variables as well